### DISCUSSION TITLE
How to get args datatypes in resolver?

### DISCUSSION THREAD

**Discussion by nengkhoiba** [apollographql/apollo-server#6122] [General]
## How to get args datatypes in resolver?

# Reason
I am building a SAAS application and Schema is generated dynamically at json config and same mutation can have different args and datatype base on the client. 

# Requirement
I want to get the datatype for the args in the resolver so that i do some custom work base on the datatype.

**Comment by glasser** [comment]
I suggest you use `graphql-code-generator` with the typescript-resolvers plugin: https://www.graphql-code-generator.com/plugins/typescript-resolvers

We are hoping to improve the docs as part of AS4 to make them TS-native including teaching how to use codegen.
